### PR TITLE
Fix linting issues in Click to Load code

### PR DESCRIPTION
**Pull request by kzar**
There were a number of ESLint and TypeScript linting issues in the
Click to Load code. Let's fix those and enable full linting for the
file.

### CODE REVIEW

**Comment by kzar** [comment]
@jonathanKingston As promised, here is a PR to get the Click to Load code passing linting. I have left this as a draft while I test I didn't break anything.

File: src/features/click-to-load.js [hunk 1]
```diff
@@ -1389,18 +1590,19 @@ export default class ClickToLoad extends ContentFeature {
         }
 
         // Listen for events from "surrogate" scripts.
-        addEventListener('ddg-ctp', (event) => {
+        addEventListener('ddg-ctp', (/** @type {CustomEvent} */ event) => {
             if (!event.detail) return
```
Comment (jonathanKingston) [inline src/features/click-to-load.js:1594]: I think this is not narrowing TS unfortunately, the following would remove the need for some of the ?'s below
```suggestion
            if (!('detail' in event)) return
```

File: src/features/click-to-load.js [hunk 2]
```diff
@@ -54,10 +67,17 @@ class DuckWidget {
         this.gatherDataElements()
         this.entity = entity
         this.widgetID = Math.random()
+        this.autoplay = false
```
Comment (jonathanKingston) [inline src/features/click-to-load.js:70]: Why was this added?

File: src/features/click-to-load.js (hunk 1 above)
Comment (kzar) [reply src/features/click-to-load.js]: Done.

File: src/features/click-to-load.js (hunk 2 above)
Comment (kzar) [reply src/features/click-to-load.js:70]: We were setting this later, which caused TypeScript to get upset that DuckWidget didn't have an `autoplay` property. Better to set it when constructing the Object instead, plus that keeps TypeScript happy when we set the property to true later.

**Review by jonathanKingston — APPROVED** [review]
LGTM thanks for this :)
